syntax = "proto3";

package currency_tracker;

service CurrencyTracker {
  rpc subscribe (Subscription) returns (stream CurrencyData);
  rpc unsubscribe (Cancelation) returns (Empty);
}

message Empty {

}

message Cancelation {
  string bankName = 1;
}

message CurrencyData {
  CurrencyName currency = 1;
  // Ratio to PLN currency
  double exchange_ratio = 2;
}

message Subscription {
  string bankName = 1;
  repeated CurrencyName currencies = 2;
}

enum CurrencyName {
  AUD = 0;
  CHF = 1;
  EUR = 2;
  GBP = 3;
  USD = 4;
}